### PR TITLE
PT-162578406 Add payable modifier

### PR DESCRIPTION
**Pull request by hanssv**
Needs https://github.com/aeternity/aebytecode/pull/69

### CODE REVIEW

**Comment by uwiger** [comment]
Still need the WIP label?

File: src/aeso_icode.erl [hunk 1]
```diff
@@ -103,6 +106,10 @@ new_env() ->
 set_name(Name, Icode) ->
     maps:put(contract_name, Name, Icode).
 
+-spec set_payable(boolean(), icode()) -> icode().
+set_payable(Name, Icode) ->
```
Comment (uwiger) [inline src/aeso_icode.erl]: `Name` is a boolean?

File: src/aeso_ast_to_icode.erl [hunk 2]
```diff
@@ -17,11 +17,15 @@
...
-               _ -> gen_error(last_declaration_must_be_contract)
+    {Payable, Name} =
+        case lists:last(TypedTree) of
+            {contract, Attrs, {con, _, Con}, _} ->
+                {proplists:get_value(payable, Attrs, false), Con};
+            _ ->
+                gen_error(last_declaration_must_be_contract)
            end,
-    NewIcode = aeso_icode:set_name(Name, aeso_icode:new(Options)),
+    NewIcode = aeso_icode:set_payable(Payable,
```
Comment (uwiger) [inline src/aeso_ast_to_icode.erl]: So here, `Payable` is definitely a boolean, but in `aeso_icode:set_payable/2`, the first argument is called `Name` (though typed as a boolean), which at least looking at this context, seems confusing.

File: src/aeso_icode.erl (hunk 1 above)
Comment (hanssv) [reply src/aeso_icode.erl]: Copy pasted, should be `Payable` of course.

**Comment by hanssv** [comment]
Now updated with the correct `aebytecode` commit

**Pull request merged**